pkg/scheduler/worker/workerpool_test.go

```diff
@@ -17,16 +20,25 @@ import (
 )
 
 type testInvoker struct {
-	params []*invoker.Params
+	params     []*invoker.Params
+	reconRepo  reconciliation.Repository
```

Comment:
a struct is a collection of data, but the data should meaningfully belong to this collection. Reconciliation repository should not belong to the invoker, and let invoker carry Mutex and Channel is also not a good solution here.

Reply:
Lets talk about it. The invoker needs the repo to update its status, as already mentioned last week in a small meeting